### PR TITLE
feat: Brillig opcode diff report for programs

### PR DESCRIPTION
**Pull request by vezenovm**
# Description

## Problem\*

Resolves 

## Summary\*

This PR simply adds similar handling for Brillig functions as already exists for normal ACIR circuits. The "unconstrained_functions" field in the info report produced by `nargo info` is now handled. Tests have been updated to account for this extra field. 

## Additional Context

- This PR only handles Noir compiled with `--force-brillig`. This was made as the goal is to profile Brillig bytecode compilation optimizations in general, not enabling developers to look at opcode sizes of individual functions like for circuits. Either way, support for multiple functions in a report needs to be added for ACIR as well. I felt this can come in a follow-up. 
- Unconstrained functions in contracts have not been updated in this PR.

# PR Checklist\*

- [X] I have tested the changes locally.
- [X] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -440,7 +440,7 @@ function run() {
                             const res = _c;
                             const artifact = res.data.find((artifact) => !artifact.expired && artifact.name === baseReport);
                             if (!artifact) {
-                                yield new Promise((resolve) => setTimeout(resolve, 900)); // avoid reaching the API rate limit
+                                yield new Promise((resolve) => setTimeout(resolve, 500)); // avoid reaching the API rate limit
```
Comment (TomAFrench) [inline dist/index.js:443]: Pretty sure that this is just making requests more frequently. We should add an optional bootstrap commit hash which we don't look further than.

Comment (vezenovm) [reply dist/index.js:443]: > Pretty sure that this is just making requests more frequently. We should add an optional bootstrap commit hash which we don't look further than.

Yeah I realized that after it was done. Guess I just lucked out the PR on noir-lang didn't hit the rate limit.

File: src/index.ts [hunk 2]
```diff
@@ -114,23 +122,82 @@ async function run() {
...
+    // `--force-brillig` then we should have a single unconstrained function marked with "main".
+    // If we compiled our entire workspace to Brillig then we can generate a Brillig report instead of a circuit report.
+    let brillig_report = false;
+    if (
+      referenceReports.programs.length > 0 &&
+      referenceReports.programs[0].unconstrained_functions.length === 1 &&
+      referenceReports.programs[0].unconstrained_functions[0].name === "main"
+    ) {
+      brillig_report = true;
+    }
```
Comment (TomAFrench) [inline src/index.ts:140]: Here we're assuming that if the first project in the workspace is unconstrained then we want an unconstrained report, no? This seems kinda fragile if we're reliant on the ordering of projects within a mixed workspace. Why not output the two reports separately similarly to how we do for shell/markdown?

Comment (vezenovm) [reply src/index.ts]: done

File: dist/index.js (hunk 1 above)
Comment (vezenovm) [reply dist/index.js]: As I got a working report now should we do this in a follow-up PR? I can do the change in this PR if we still desire that though.

**Review by TomAFrench — APPROVED** [review]

**Pull request merged**